server/links/datatrails/readme.md

```diff
@@ -0,0 +1,253 @@
+2. Install the required dependencies:
+
+   ```bash
+   pip install requests
+   ```
+
+## Usage
+
+The DataTrails Link will automatically process vCons as they pass through the conserver providing integrity protection to the "moments that matter".
+
+### Ledgering a vCon
+
+For each vCon Chain:
+
+1. A [DataTrails Event]() will be recorded, integrity protecting each vCon, setting the `subject` to `vcon://<vcon_uuid>` for correlation of events to each vCon
+1. If no DataTrails Asset exists for the vCon, a new asset will be created.  
+   _**Note:** this is a temporary solution as Assets are being deprecated._
+
+## Configuration
+
+The DataTrails Link requires minimal configuration in the conserver setup.
+
+- Add the following datatrails configurations to the conserver configuration file (`config.yml`)
+- Replace `"<your_client_id>"` and `"<your_client_secret>"` with your [DataTrails API credentials](https://docs.datatrails.ai/developers/developer-patterns/getting-access-tokens-using-app-registrations/)
+- Set `vcon_operation` to differentiate what operation is being recorded on DataTrails/SCITT.  
+- The DataTrails `arc_display_type` is configured as: `"vcon_operation}"`, enabling [configurable permissions on Event Types](https://docs.datatrails.ai/platform/administration/sharing-access-outside-your-tenant/#creating-an-obac-policy)  
+
+```yaml
+links:
+  datatrails-create:
+    module: links.datatrails
+    options:
+      datatrails_client_id: "<your_client_id>"
+      datatrails_client_secret: "<your_client_secret>"
+      vcon_operation: "vcon-create"
+  datatrails-consent:
+    module: links.datatrails
+    options:
+      datatrails_client_id: "<your_client_id>"
+      datatrails_client_secret: "<your_client_secret>"
+      vcon_operation: "vcon-consent"
+
+chains:
+  create_chain:
+    links:
+      - datatrails-create
+    ingress_lists:
+      - create_ingress
+    egress_lists:
+      - default_egress
+    enabled: 1
+  consent_chain:
+    links:
+      - datatrails-consent
+    ingress_lists:
+      - consent_ingress
+    egress_lists:
+      - default_egress
+    enabled: 1
+```
+
+## Verifying vCons
+
+DataTrails provides several APIs for verifying the integrity and inclusion of changes to a vCons history.
+We'll also explore specific vCon scenarios, such as consent and revocation validation.
+
+### Retrieving All vCon Events
+
+For each important operation performed on a vCon, a DataTrails Event (SCITT Signed Statement) should be recorded.
+
+To align with SCITT semantics, the vcon_uuid is set to the DataTrails `subject` event attribute. (`event_attributes.subject`)
+
+To query the history of DataTrails Events for a given vCon, use the following:
+
+- For bash/curl commands, configure the `.datatrails/bearer-token.txt` using the DataTrails [Creating Access Tokens](https://docs.datatrails.ai/developers/developer-patterns/getting-access-tokens-using-app-registrations/) developer docs.
```

Comment:
Ensure that the .datatrails/bearer-token.txt file follows the ssh convention on permissions:

http://jonasbn.github.io/til/ssh/permissions_on_ssh_folder_and_files.html

~/.datatrails 700   (drwx------)
~/.datatrails/bearer-token.txt  600 (-rw-------)

Reply:
we removed the file writing, so this shouldn't be an issue any longer.